pkg/controller/workload/workload_processor.go

```diff
@@ -682,10 +702,24 @@ func (p *Processor) updateServiceMap(service *workloadapi.Service) error {
 				return nil
 			}
 		}
+
+		// Compare the addresses of the old and new maps to avoid residual.
+		// If the data can be found in the km_service map, it is also stored in the serviceCache.
+		newServiceAddress := utils.GetAddressesFromService(service)
+		oldServiceAddress := utils.GetAddressesFromService(oldService)
+		_, removeServiceAddress := utils.CompareIpByte(newServiceAddress, oldServiceAddress)
+
+		frontKey := bpf.FrontendKey{}
+		for _, address := range removeServiceAddress {
+			nets.CopyIpByteFromSlice(&frontKey.Ip, address)
+			if err := p.bpf.FrontendDelete(&frontKey); err != nil {
```

Comment:
we can abstract a new function as i see it is used by workload handler too

Reply:
Done